feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/input/ui/DateOfOccurenceAndLocation.kt

```diff
@@ -0,0 +1,86 @@
+package com.hedvig.android.odyssey.input.ui
+
+import android.app.DatePickerDialog
+import android.widget.DatePicker
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.component.button.FormRowButton
+import com.hedvig.android.core.designsystem.component.button.LargeContainedTextButton
+import com.hedvig.android.odyssey.model.ClaimState
+import com.hedvig.android.odyssey.repository.AutomationClaimDTO2
+import java.time.LocalDate
+
+@Composable
+fun DateOfOccurrenceAndLocation(
+  state: ClaimState,
+  onDateOfOccurrence: (LocalDate) -> Unit,
+  locationOptions: List<AutomationClaimDTO2.ClaimLocation>,
+  onLocation: (AutomationClaimDTO2.ClaimLocation) -> Unit,
+  onNext: () -> Unit,
+) {
+  val openLocationPickerDialog = remember { mutableStateOf(false) }
+
+  val now = LocalDate.now()
+  val pickerDialog = DatePickerDialog(
+    LocalContext.current,
+    { _: DatePicker, year: Int, month: Int, dayOfMonth: Int ->
+      onDateOfOccurrence(LocalDate.of(year, month, dayOfMonth))
+    },
+    now.year,
+    now.monthValue,
+    now.dayOfMonth,
+  )
```

Comment:
Good catch. Can we use the newer date picker (from m3?) instead? Perhaps in another PR?

Reply:
Yes, keep it all as it is, and when you feel like it can be fitting, I can do a sweep and change all of them to the m3 picker in one PR. I can start doing that for example after we've done the first merge to develop and have it all feature flagged and we can iteratively make it reach the point where we can remove the flag